lib/ansible/modules/cloud/azure/azure_rm_aks.py

```diff
@@ -0,0 +1,537 @@
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            dns_prefix=dict(
+                type='str',
+                required=True
+            ),
+            kubernetes_version=dict(
+                type='str',
+                required=False
+            ),
+            linux_profile=dict(
+                type='list',
+                required=True
+            ),
+            agent_pool_profiles=dict(
+                type='list',
+                required=True
+            ),
+            service_principal=dict(
+                type='list',
+                required=False
+            ),
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.dns_prefix = None
+        self.kubernetes_version = None
+        self.tags = None
+        self.state = None
+        self.linux_profile = None
+        self.agent_pool_profiles = None
+        self.service_principal = None
+
+        self.results = dict(changed=False, state=dict())
+
+        super(AzureRMManagedCluster, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                    supports_check_mode=True,
+                                                    supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        resource_group = None
+        response = None
+        results = dict()
+        to_be_updated = False
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        # Check if the AKS instance already present in the RG
+        if self.state == 'present':
+
+            if not self.service_principal:
+                self.fail('service_principal should be specified when using AKS')
+            if not self.service_principal[0].get('client_id'):
+                self.fail('service_principal.client_id should be specified when using AKS')
+            if not self.service_principal[0].get('client_secret'):
+                self.fail('service_principal.client_secret should be specified when using AKS')
+
+            # For now Agent Pool cannot be more than 1, just remove this part in the future if it change
+            agentpoolcount = len(self.agent_pool_profiles)
+            if agentpoolcount > 1:
+                self.fail('You cannot specify more than agent_pool_profiles')
```

Comment:
more than one?

Reply:
I think this logic should be handled by service instead of ansible client side

Reply:
API allows creation of more than one agent pool profile, but it won't provision correctly iirc. This is a workaround until API is fixed.

Reply:
The same as the linux profile update?